dask_expr/expr.py

```diff
@@ -1173,6 +1174,38 @@ class Drop(Elemwise):
     operation = staticmethod(drop_by_shallow_copy)
 
 
+class _SetPartitionsPreSetIndex(Blockwise):
+    _parameters = ["frame", "new_divisions", "ascending", "na_position"]
+    _defaults = {"ascending": True, "na_position": "last"}
+    operation = staticmethod(set_partitions_pre)
+
+    @property
+    def _meta(self):
+        return self.frame._meta._constructor([0])
+
+
+class _SetIndexPostScalar(Blockwise):
+    _parameters = ["frame", "index_name", "drop"]
+
+    def operation(self, df, index_name, drop):
+        df2 = df.set_index(index_name, drop=drop)
+        return df2
+
+
+class _SetIndexPostSeries(Blockwise):
+    _parameters = ["frame", "index_name"]
+
+    def operation(self, df, index_name):
+        df2 = df.set_index("_index", drop=True)
+        df2.index.name = index_name
+        return df2
+
+
+class SortIndexBlockwise(Blockwise):
+    _parameters = ["frame"]
+    operation = M.sort_index
+
+
```

Comment:
These should maybe stay close to the set_index implementation in `shuffle.py`?

Reply:
Oh that makes sense, thx.

Reply:
Is there a reason why this one survived?  I could imagine two reasons:
1.  At some point we're thinking about `sort=False`.
2. Copy-paste mistake

In either case I think it probably makes sense to move.  It isn't really related to any other code in that this file.

Reply:
This is more broad for me, like has other applications than ``set_index`` only. It's probably better to move this for now as well and move back in case we need it at some point